src/pybind/cephfs/cephfs.pyx

```diff
@@ -1901,6 +1901,23 @@ cdef class LibCephFS(object):
                           st_mtime=datetime.fromtimestamp(stx.stx_mtime.tv_sec),
                           st_ctime=datetime.fromtimestamp(stx.stx_ctime.tv_sec))
 
+    def is_dir_exists(self, path):
+        try:
+            return (self.stat(os.path.join(b'', path)).st_mode &
+                    0o0040000) != 0
+        except ObjectNotFound:
+            return False
+
+    def is_file_exists(self, path):
+        try:
+            return (self.stat(os.path.join(b'', path)).st_mode &
+                    0o0040000) == 0
+        except ObjectNotFound:
+            return False
```

Comment:
It seems there is no difference between `is_dir_exists()` and `is_file_exists()`. Could we just fold them into a single func ? Or just do:

```
def is_path_exists(self, path):
        try:
            return (self.stat(os.path.join(b'', path)).st_mode &
                    0o0040000) == 0
        except ObjectNotFound:
            return False
```

Reply:
Ah this is not required anymore, first commit will be removed.

Reply:
Okay.